htsinfer/get_library_source.py

```diff
@@ -71,16 +75,59 @@ def evaluate(self) -> ResultsSource:
             Source results object.
         """
         source = ResultsSource()
-        index = self.create_kallisto_index()
-        source.file_1 = self.get_source(
-            fastq=self.paths[0],
-            index=index,
-        )
-        if self.paths[1] is not None:
-            source.file_2 = self.get_source(
-                fastq=self.paths[1],
+        # Check if library_source is provided, otherwise infer it
+        if self.org_id is not None:
+            source.file_1.taxon_id = self.org_id
+            org_name = self.get_organism_name(
+                self.org_id,
+                self.transcripts_file
+            )
+
+            if org_name is not None:
+                source.file_1.short_name = org_name
+
+                if self.paths[1] is not None:
+                    source.file_2.taxon_id = self.org_id
+                    source.file_2.short_name = source.file_1.short_name
+
+            else:
```

Comment:
So in a previous version, I implemented what you mentioned (stop the run when the tax ID is invalid or not found in transcript list), then changed it to just a warning, as my thinking was that the program should return _some_ result when it's run, but then the invalid tax ID with no corresponding source name would get reported back as the "inferred" source at the end of the run, and that didn't seem right. 

It's true that there is no difference in terms of results between the run with not supported tax ID and when inferring but no organism is found in the transcript list, but maybe there could be cases when the inferred org was actually what the user wanted in the first place, so inferring it despite given a wrong tax ID could be beneficial? 

For example, in our transcript list we have "Oryza sativa" with tax ID 39947, but when looking it up in the NCBI db, it turns out it's actually [Oryza sativa Japonica Group](https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=Info&id=39947&lvl=3&lin=f&keep=1&srchmode=1&unlock), and when searching for the tax ID for just "Oryza sativa", it's actually [4530](https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=Info&id=4530&lvl=3&lin=f&keep=1&srchmode=1&unlock). I know this is a very niche case, or maybe just a quirk from how the transcript list was collected from different subspecies and whatnot, so to me it made sense to report back a result with an inferred source, even when a wrong tax ID was an input. 

On second thought though, this could also backfire, if the inferred org is something totally different from the given tax ID (like giving an ID for some mouse subspecies that's not in our list, but a butterfly gets inferred), so throwing an error would be probably the best 😅 I'll refactor it.

Reply:
Yes, there are some issues from the sourcing of the transcripts for the different species. It was, in fact, quite painful to prepare the list of source names and tax IDs, because there are actually quite a lot of inconsistencies.

Still, I think it's safer not to just do something that the user didn't ask for. If you supply the source via the tax ID, and we _know_ the tax ID, then we assume the user is correct - even if it's incorrect. Not our problem. And if we don't know the tax ID, we just tell them, even if they provided a tax ID for a source that actually _is_ supported - but under a different tax ID. If anything, we should change this by working on the sourcing of the genome resources, _not_ by adapting our program logic to it.

Now, if you are concerned about the results _suggesting_ that we inferred something that was actually given as an input, you could add a field to the results that mentions that the source ID was given as an input. Or better yet, a field that lists _all_ of the result fields that were given as an input (e.g., the tax ID), or in another way trivially derived from an input (e.g., the source name when the tax ID is given) - in case we also later allow users to provide other inputs to avoid inference.